src/telliot/reporter/interval.py

```diff
@@ -23,76 +25,118 @@ def __init__(
-                        elif (
-                            not status.ok
-                            and status.error
-                            and "replacement transaction underpriced" in status.error
-                        ):
-                            extra_gas_price += gas_price
-                        else:
-                            extra_gas_price = 0
+        status = ResponseStatus()
+        gas_price_gwei = await fetch_gas_price()
 
-                else:
-                    print(
-                        f"Skipping submission for {repr(datafeed)}, "
-                        f"no query for datafeed."
-                    )  # TODO logging
+        transaction_receipts: List[
+            Tuple[Optional[AttributeDict[Any, Any]], ResponseStatus]
+        ] = []
+
+        user = self.endpoint.web3.eth.account.from_key(self.private_key).address
+        is_staked, read_status = await self.master.read("getStakerInfo", _staker=user)
+
+        if (not read_status.ok) or (is_staked is None):
+            status.error = "unable to read reporter staker status: " + read_status.error  # type: ignore # error won't be none # noqa: E501
+            status.e = read_status.e
+            transaction_receipts.append((None, status))
+
+        else:
+            print(is_staked[0])
+
+            # Status 1: staked
+            if is_staked[0] == 1:
+                jobs = []
+                for datafeed in self.datafeeds:
+                    job = asyncio.create_task(datafeed.source.fetch_new_datapoint())
+                    jobs.append(job)
+
+                _ = await asyncio.gather(*jobs)
+
+                for datafeed in self.datafeeds:
+
+                    datapoint = datafeed.source.latest
+                    v, t = datapoint
+
+                    if v is not None:
+                        query = datafeed.query
+
+                        if query:
+                            value = query.value_type.encode(v)
+                            query_id = query.query_id
+                            query_data = query.query_data
+                            extra_gas_price = 20
+
+                            timestamp_count, read_status = await self.oracle.read(
+                                func_name="getTimestampCountById", _queryId=query_id
+                            )
+
+                            if not read_status.ok:
+                                status.error = "unable to retrieve timestampCount: " + read_status.error  # type: ignore # error won't be none # noqa: E501
+                                status.e = read_status.e
+                                transaction_receipts.append((None, status))
+
+                            tx_receipt, status = await self.oracle.write_with_retry(
+                                func_name="submitValue",
+                                gas_price=gas_price_gwei,
+                                extra_gas_price=extra_gas_price,
+                                retries=5,

```

Comment:
Arbitrary number or was three retries too low?

Reply:
For now, I use 5 to sus out new rpc errors from the tox console :)